component/common/relabel/relabel.go

```diff
@@ -184,3 +184,10 @@ func ComponentToPromRelabelConfigs(rcs []*Config) []*relabel.Config {
 
 	return res
 }
+
+// Rules returns the relabel configs in use for a relabeling component.
+type Rules func() []*relabel.Config
```

Comment:
w/r/t returning the Flow vs Prometheus type:

The Flow type probably makes more sense, but if it's going to be too much work, this is fine for now.

Reply:
I agree with this take.
The components weren't keeping track of the Flow variant so I've added this, but I believe it makes more sense for inter-component communication.